### PR TITLE
Fix clusterctl workflow

### LINKED ISSUE #183: clusterctl flow is currently broken for Ubuntu
When following the documentation in README.md, `clusterctl` will not work, because of missing ssh keys.

When manually adding those, and reference them in the `machines.yaml`, the nodes come up, but the `docker.service` does not include `--iptables=false and --ip-masq=false`.

Also the control plane does not come up, because of a missing `controlPlaneEndpoint`.

### PR DESCRIPTION
**Pull request by chrigl**
**What this PR does / why we need it**:

With these changes in, it is possible to follow `README.md` and have a running ubuntu cluster afterwards.

...

Fixed rendering of userData

additionally added BatchMode to ssh, because I discovered a password
prompt during bringing up a centos machine. This must just fail.

Fixed inserting of cloud.conf in generate-yaml.sh

otherwise, cloud.conf does not end up on the nodes, but are referenced
in config files and thus, services do not come up.

Fixed userdata scripts and machines.yaml

Included default keyName to machines

Fixed docker systemd.service to handle DOCKER_OPTS

Included public-ipv4 as controlPlaneEndpoint to make bootstrap possible
by just editing machines.yaml

Extended documentation to bring up a cluster

Added additional requirement to manually upload an ssh key

Included ssh-key and controlPlaneEndpoint to CentOS

Even though, centos does not work in my tests so far.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #183 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/cloud/openstack/machine/actuator.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 	}
 
 	var userDataRendered string
-	if len(userData) == 0 {
+	if len(userData) > 0 {
```
Comment (flaper87) [inline pkg/cloud/openstack/machine/actuator.go]: jeeeeeez! Can't believe I messed this `if` statement.

File: README.md [hunk 2]
```diff
@@ -86,9 +86,18 @@ Participation in the Kubernetes community is governed by the [Kubernetes Code of
...
 
-   You will need to make changes to the generated files to create a working cluster.
+   You **will need** to make changes to the generated files to create a working cluster.
    You can find some guidance on what needs to be edited, and how to create some of the
-   required OpenStack resources in the [Configuration documentation](docs/config.md)
+   required OpenStack resources in the [Configuration documentation](docs/config.md).
+
+   #### Special notes on ssh keys and fetching `admin.conf`
+
+   When running `generate-yaml.sh` the first time, a new ssh keypair is generated and stored as `$HOME/.ssh/openstack_tmp` and `$HOME/.ssh/openstack_tmp.pub`. In order to allow `clusterctl` to fetch kubernetes `admin.conf` from the master node, you **must** manually create it in OpenStack. By default the generated `machine.yaml` uses `cluster-api-provider-openstack` to be the `keyName`. However, you are free to change that.
```
Comment (chaosaffe) [inline README.md]:
```suggestion
   When running `generate-yaml.sh` the first time, a new ssh keypair is generated and stored as `$HOME/.ssh/openstack_tmp` and `$HOME/.ssh/openstack_tmp.pub`. In order to allow `clusterctl` to fetch Kubernetes' `admin.conf` from the master node, you **must** manually create the key pair in OpenStack. By default the generated `machine.yaml` uses `cluster-api-provider-openstack` to be the `keyName`. However, you are free to change that.
```

**Comment by chaosaffe** [comment]
Reiterate /lgtm from @flaper87

**Pull request merged**